### ISSUE TITLE
Multiple Update, Multiple SoftDelete, Hard Delete, Many To Many relation issue

### ISSUE THREAD

**Issue by khalilst** [VentureCraft/revisionable#305]
Revisinable isn't working with below commands:

`User::whereIn('id', [1,2,3,...])->update(['some_field' => 'some_value']); //Multiple Update`
`User::whereIn('id', [1,2,3,...])->delete(); //Multiple soft-deleteThe User is soft-deletable`
`User::find(1)->roles()->sync([1,3,5,...]); //Many to Many Relations`

Also it would be nice to record hard-delete in revisions table.
I record revision for all above methods manually. But it doesn't feel good :(

**Comment by fico7489** [comment]
@khalilst  how you propose to track hard-deletes?

**Comment by khalilst** [comment]
Tracking hard delete indicates a user with an ID delete a record with an ID
and specific type.
I think it could be useful.

**Comment by fico7489** [comment]
yes of yours it would be useful but I am asking how you think to save this?

You have this schema 

```
        Schema::create('revisions', function ($table) {
            $table->increments('id');
            $table->string('revisionable_type');
            $table->integer('revisionable_id');
            $table->integer('user_id')->nullable();
            $table->string('key');
            $table->text('old_value')->nullable();
            $table->text('new_value')->nullable();
            $table->timestamps();

            $table->index(['revisionable_id', 'revisionable_type']);
        });
```

what should we put in revisionable_type, revisionable_id, old_value and new_value fields for hard delete ?

**Comment by khalilst** [comment]
All fields similar to creating a record except key to 'id' for example and
old_value to current id and new_value to null.
